### PR TITLE
Add arguments to JkInst

### PR DESCRIPTION
**Pull request by n1tram1**

### CODE REVIEW

File: src/instruction/mod.rs [hunk 1]
```diff
@@ -132,3 +132,20 @@ impl Clone for Box<dyn Instruction> {
...
+
+pub struct CodeGen<'ctx> {
+    context: &'ctx Context,
+    module: Module<'ctx>,
+    builder: Builder<'ctx>,
+}
+
+pub trait Compilable {
+    fn ir(&self, ctx: &mut CodeGen, interpreter: &Interpreter) -> Result<String, JkError>;
+}
```
Comment (CohenArthur) [inline src/instruction/mod.rs:151]: I think this is a bit outside the scope of this PR but that's okay. However, careful of unused code.

The issue is that I believe this code should reside in its own file, but I'm not sure of what architecture to use. I think using another module would be best

Comment (n1tram1) [reply src/instruction/mod.rs:151]: oops this shouldn't have been pushed

File: src/instruction/jk_inst.rs [hunk 2]
```diff
@@ -36,23 +55,56 @@ impl Instruction for JkInst {
...
+        assert!(inst.is_ok(), "dump is a valid interpreter directive")
+    }
+
+    #[test]
+    fn t_valid_inst_with_args() {
+        let (_, fc) = Construct::function_call("ir(fn)").unwrap();
+        let inst = JkInst::from_function_call(fc);
+
+        assert!(inst.is_ok(), "ir(func) is a valid use of the ir interpreter directive")
+    }
```
Comment (Skallwar) [inline src/instruction/jk_inst.rs:109]: No test for ``dump(stuff_here)`` or ``ir()``

**Review by Skallwar — APPROVED** [review]

File: src/instruction/jk_inst.rs (hunk 2 above)
Comment (n1tram1) [reply src/instruction/jk_inst.rs:109]: They are both valid syntax and all interpreter directives are variadic, so I don't know if it should be tested.

File: src/instruction/jk_inst.rs [hunk 3]
```diff
@@ -3,30 +3,49 @@
...
                 None,
-                keyword.to_owned(),
+                func_name.to_owned(),
             )),
-        }
+        };
+
+        Ok(Self {
+            kind,
+            args: fc.args().to_owned()
```
Comment (CohenArthur) [inline src/instruction/jk_inst.rs:43]:
```suggestion
            args: fc.args().clone()
```
I think the behavior is the same but I might be mistaken. If it is the same, then I think this version is clearer and shows the intent better

**Review by CohenArthur — APPROVED** [review]
Thanks!

**Review by CohenArthur — APPROVED** [review]

**Review by Skallwar — APPROVED** [review]
There is still some IR stuff but ok

File: src/instruction/var_assignment.rs [hunk 4]
```diff
@@ -23,10 +23,15 @@ impl VarAssign {
         }
     }
 
-    /// Get a reference to the symbol of the variable declaration
+    /// Get a reference to the symbol of the variable assignment
     pub fn symbol(&self) -> &str {
         &self.symbol
     }
+    ///
```
Comment (CohenArthur) [inline src/instruction/var_assignment.rs:30]:
```suggestion
```

**Pull request merged**